### ISSUE TITLE
Encoding errors in mails

### ISSUE THREAD

**Issue by dmolineus** [contao/contao#3484]
**Affected version(s)**

Contao 4.9 and probably Contao 4.12 as well

**Description**

I have contact form and use the send mail feature of the Contao Core. Special chars, like single quotes get encoded in the text mail.

Expected:

```
Name: Test
Mail: test@example.org
Nachricht: I'm a bug hunter
Test: foo, bar
```

Result:
```
Name: Test
Mail: test@example.org
Nachricht: I&#039;m a bug hunter
Test: foo, bar
```

**Comment by fritzmg** [comment]
I think this is already fixed in Contao `4.9.x-dev`. However, I did notice another encoding problem, when using the `{{post::*}}` insert tag in the email subject for instance:

```
Content-Transfer-Encoding: quoted-printable
Content-Type: text/plain; charset=utf-8
Date: Tue, 21 Sep 2021 08:29:22 +0100
From: foobar@example.com
MIME-Version: 1.0
Message-ID: <8fc8dd82d9cd41c3559b1b0be129493c@c49.local>
Received: from c49.local by mailhog.example (MailHog)

          id GQApr4Xl2RfCELKPKH8l-XQHluI2CFsbEsjWRlOGWlY=@mailhog.example; Tue, 21 Sep 2021 08:29:22 +0100
Return-Path: <foobar@example.com>,<foobar@example.com>
Subject: Email from c49.local: I&#39;m a developer
To: foobar@example.com

Name: I'm a developer
```

The configured email subject is `Email from {{env::host}}: {{post::name}}`, but the subject will be

```
Email from c49.local: I&#39;m a developer
```

/cc @ausi

**Comment by ausi** [comment]
This needs to be handled in https://github.com/contao/contao/blob/c88eddcf7eb71916036512d21ebbe53dfdcb1eed/core-bundle/src/Resources/contao/forms/Form.php#L405 I think because insert tags always assume to be used within HTML context.

The String `I <3 Contao` would have always had this issue I guess? (even before the security fixes)

**Comment by fritzmg** [comment]
> I think because insert tags always assume to be used within HTML context.

Agreed 👍 

> The String `I <3 Contao` would have always had this issue I guess? (even before the security fixes)

Yep 👍. `4.9.17`:

```
Content-Transfer-Encoding: quoted-printable
Content-Type: text/plain; charset=utf-8
Date: Tue, 21 Sep 2021 09:46:16 +0100
From: foobar@example.com
MIME-Version: 1.0
Message-ID: <32bae2565a0ae8ce37e93f5db702f50f@c49.local>
Received: from c49.local by mailhog.example (MailHog)

          id vyv9nDEZCj96JQkEsNGnEFwVcQjzfMlrnmU_Jc6jADA=@mailhog.example; Tue, 21 Sep 2021 09:46:16 +0100
Return-Path: <foobar@example.com>,<foobar@example.com>
Subject: Email from c49.local: I &lt;3 Contao
To: foobar@example.com

Name: I <3 Contao
```

**Comment by ausi** [comment]
Closed in favor of #3512
